### PR TITLE
Feature/1 yum centos7 agent

### LINKED ISSUE #1: Create a YUM CentOS 7 agent

### PR DESCRIPTION
**Pull request by OrBin**
Closes #1

### CODE REVIEW

File: yum-centos7/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+flask == 1.1.1
+typing == 3.5.2.2
+setuptools >= 21.0.0
+six == 1.13.0
```
Comment (JackZafrani) [inline yum-centos7/requirements.txt:4]: shouldn't yum be here as well?

**Review by JackZafrani — COMMENTED** [review]
I tried to play with it a bit. If i omit the architecture type i get back HTTP status 200 (OK). I think that in the case of yum that should be a bad request (likewise for os variant)

File: yum-centos7/requirements.txt (hunk 1 above)
Comment (OrBin) [reply yum-centos7/requirements.txt:4]: No. It cannot be installed with pip, and is provided as a system package, with the built-in Python installation.

**Comment by OrBin** [comment]
> I tried to play with it a bit. If i omit the architecture type i get back HTTP status 200 (OK). I think that in the case of yum that should be a bad request (likewise for os variant)

I agree, fixed. About OS variant, I'd like to take it offline for discussion.

File: yum-centos7/swagger_server/controllers/downloads_controller.py [hunk 2]
```diff
@@ -43,23 +51,38 @@ def _download_packages(self, patterns):
...
+
+        compress_files(req.compressed_file_type,
+                       package_local_paths,
+                       self.__processed_requests[request_id].output_file_path,
+                       password=None)
+
+        req.status = 'Ready'
+        req.is_consumable = True
+
+        self.__scheduler.enter(self.REQUEST_KEEPING_TIMEOUT_SEC, 1, self._delete_request, (request_id,))
```
Comment (JackZafrani) [inline yum-centos7/swagger_server/controllers/downloads_controller.py:83]: Are we sure that it is a long enough TTL?

**Review by JackZafrani — CHANGES_REQUESTED** [review]
@OrBin We need to change the way we search for packages as we discussed offline.

**Review by JackZafrani — APPROVED** [review]
It looks good. 
if i understood this properly, when a download is requested if there is no "spec" in the body, HTTP status 500 will be returned.

File: yum-centos7/swagger_server/controllers/downloads_controller.py (hunk 2 above)
Comment (OrBin) [reply yum-centos7/swagger_server/controllers/downloads_controller.py:83]: I believe it is for now. We can always change it later :)

**Comment by OrBin** [comment]
> It looks good.
> if i understood this properly, when a download is requested if there is no "spec" in the body, HTTP status 500 will be returned.

You are right, nice catch!
I'm fixing it in the coordinator.

**Pull request merged**